staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -180,6 +158,72 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	return patch, patchedObj, err
 }
 
+// getPatchType returns correct PatchType for the given GVK.
+func (p *Patcher) getPatchType(gvk schema.GroupVersionKind) (types.PatchType, error) {
```

Comment:
Yeah, I'm not super convinced yet, now you have the logic deciding what type of patch that is slightly split into two locations: This, and some of the logic is above, I think it's worse.

Reply:
Sorry, I think I am missing something. What my plan is after two PR are merged, this function will look like this https://github.com/kubernetes/kubernetes/pull/110454/files#diff-5dbe477b3463f52cdf1d6913df2f8f3e03907f37306d90e85d4e5386ca6ee0e6R164-R186

Reply:
yeah, for now I don't really mind what it will look like in the future :-) I just don't think this looks great because the logic is split, IMHO, in a weird way.

Reply:
Fair enough:). I'll move this logic back to main function

Reply:
Thanks, maybe I won't like it lol sorry about that :-)